modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenClientCredentialsGrantTestCase.java

```diff
@@ -0,0 +1,354 @@
+    }
+
+    @Test(groups = "wso2.is", description = "Verify the replacement of the 'expires_in' claim in the access token")
+    public void testTokenExpiresInClaimReplaceOperation() throws Exception {
+
+        if (jwtClaims.getClaim("expires_in") != null) {
+            Object expValue = jwtClaims.getLongClaim("expires_in");
+            Assert.assertEquals(expValue, 7200);
+        }
+    }
+
+    /**
+     * Creates an action for pre-issuing an access token with basic authentication.
+     */
+    private String createPreIssueAccessTokenAction() {
+
+        AuthenticationType authenticationType = new AuthenticationType();
+        authenticationType.setType(AuthenticationType.TypeEnum.BASIC); // todo handle mock server authorization
+        Map<String, Object> authProperties = new HashMap<>();
+        authProperties.put(USERNAME_PROPERTY, TEST_USER);
+        authProperties.put(PASSWORD_PROPERTY, ADMIN_WSO2);
+        authenticationType.setProperties(authProperties);
+
+        Endpoint endpoint = new Endpoint();
+        endpoint.setUri(EXTERNAL_SERVICE_URI);
+        endpoint.setAuthentication(authenticationType);
+
+        ActionModel actionModel = new ActionModel();
+        actionModel.setName("Access Token Pre Issue");
+        actionModel.setDescription("This is a test pre issue access token type");
+        actionModel.setEndpoint(endpoint);
+
+        try {
+            return createAction(PRE_ISSUE_ACCESS_TOKEN_API_PATH, actionModel);
+        } catch (IOException e) {
+            throw new RuntimeException("Error while creating pre issue access token " + actionModel.getName());
+        }
+    }
+
+    /**
+     * Retrieves an access token for the application.
+     *
+     * @param applicationId ID of the application
+     * @param customScopes  Custom scopes related to the integrated domain APIs
+     * @return Access token
+     * @throws Exception If error occurred wile requesting access token
+     */
+    private String retrieveAccessToken(String applicationId, List<String> customScopes) throws Exception {
+
+        OpenIDConnectConfiguration oidcConfig = getOIDCInboundDetailsOfApplication(applicationId);
+        clientId = oidcConfig.getClientId();
+        String tenantedTokenURI = getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT, tenantInfo.getDomain());
+
+        List<Permission> permissions = new ArrayList<>();
+        Collections.addAll(permissions,
+                new Permission(INTERNAL_ORG_USER_MANAGEMENT_LIST),
+                new Permission(INTERNAL_ORG_USER_MANAGEMENT_VIEW),
+                new Permission(INTERNAL_ORG_USER_MANAGEMENT_CREATE),
+                new Permission(INTERNAL_ORG_USER_MANAGEMENT_UPDATE),
+                new Permission(INTERNAL_ORG_USER_MANAGEMENT_DELETE)
+                          );
+        customScopes.forEach(scope -> permissions.add(new Permission(scope)));
+
+        return requestAccessToken(clientId, oidcConfig.getClientSecret(), tenantedTokenURI,
```

Comment:
Isn't this method get the access token using password grant type ?

Reply:
Updated with the commit 3f3b2671fc4aac10f5292af1606f409f6b30b2e9